xclim/ensembles/_robustness.py

```diff
@@ -328,48 +328,57 @@ def change_significance(  # noqa: C901
 
 def robustness_categories(
     changed_or_fractions: xr.Dataset | xr.DataArray,
-    agree: xr.DataArray = None,
+    agree: xr.DataArray | None = None,
     *,
-    categories: list[str] = [
-        "Robust signal",
-        "No change or no signal",
-        "Conflicting signal",
-    ],
-    ops: list[tuple[str, str]] = [(">=", ">="), ("<", None), (">=", "<")],
-    thresholds: list[tuple[float, float]] = [(0.66, 0.8), (0.66, None), (0.66, 0.8)],
+    categories: list[str] | None = None,
+    ops: list[tuple[str, str]] | None = None,
+    thresholds: list[tuple[float, float]] | None = None,
```

Comment:
It's not so much that it's complex values, it's because those values are treated as mutable objects. It's very easy for a user to accidentally modify the default arguments during calls, leading to weird behaviour when called again.

https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument